src/main/java/org/broadinstitute/consent/http/resources/DataSetResource.java

```diff
@@ -57,16 +62,43 @@
     private final String END_OF_LINE = System.lineSeparator();
     private final String TSV_DELIMITER = "\t";
     private final DataSetAPI api;
+    private final DatasetService datasetService;
     private final DataAccessRequestAPI dataAccessRequestAPI;
     private final UserService userService;
 
     @Inject
-    public DataSetResource(UserService userService) {
+    public DataSetResource(DatasetService datasetService, UserService userService) {
         this.dataAccessRequestAPI = AbstractDataAccessRequestAPI.getInstance();
         this.api = AbstractDataSetAPI.getInstance();
+        this.datasetService = datasetService;
         this.userService = userService;
     }
 
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("/v2")
+    @PermitAll
+    public Response createDataset(@Auth AuthUser user, String json) {
+        DataSetDTO ds = new Gson().fromJson(json, DataSetDTO.class);
+        if (ds == null) {
+            throw new BadRequestException("Dataset is required");
+        }
+        if (ds.getProperties() == null) {
+            throw new BadRequestException("Dataset must contain required properties");
+        }
+        String name = ds.getPropertyValue("Dataset Name");
+        if (name == null) {
+            throw new BadRequestException("Dataset name is required");
+        }
+        DataSet datasetNameAlreadyUsed = datasetService.findDatasetByName(name);
+        if (Objects.nonNull(datasetNameAlreadyUsed)) {
+            throw new NotFoundException("Dataset name: " + name + " is already in use");
+        }
+        DataSetDTO dataset = datasetService.createDataset(ds, name);
+        return Response.ok().entity(dataset).status(201).build();
```

Comment:
Optional. If you call the created method, it will have the right status. Unfortunately, it also requires a URI to describe the location of the new dataset. That could be pretty simple though. Add in a `@Context UriInfo info` to the method signature and then change these lines:
```suggestion
        URI uri = info.getRequestUriBuilder().replacePath("api/dataset/{datasetId}").build(ds.getDataSetId());
        return Response.created().entity(dataset).build();
```
Caveat ... untested, but this should work out of the box.